bench/bm_case1.cpp

```diff
@@ -89,13 +174,84 @@ class copy : public fg::node<copy<T, N_MIN, N_MAX, use_bulk_operation, use_memco
     }
 };
 
-template<typename T1, typename T2, std::size_t N_MIN = 0, std::size_t N_MAX = N_MAX>
-class convert : public fg::node<convert<T1, T2, N_MIN, N_MAX>, fg::IN<T1, "in", N_MIN, N_MAX>, fg::OUT<T2, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
+
+namespace detail {
+    template<typename T>
+    constexpr std::size_t simd_size() noexcept {
+        namespace stdx = vir::stdx;
+        if constexpr (stdx::is_simd_v<T>) {
+            return T::size();
+        } else {
+            return 1LU;
+        }
+    }
+}
+
+namespace stdx = vir::stdx;
+
+template<typename T1, typename T2, std::size_t N_MIN = 0 /* SIMD size */, std::size_t N_MAX = N_MAX>
+class convert
+        : public fg::node<convert<T1, T2, N_MIN, N_MAX>, fg::IN<T1, "in", N_MIN, N_MAX>, fg::OUT<T2, "out", N_MIN, N_MAX>, fg::limits<N_MIN, N_MAX>> {
+    static_assert(stdx::is_simd_v<T1> != stdx::is_simd_v<T2>, "either input xor output must be SIMD capable");
+    constexpr static std::size_t N_INPUT_SIMD_SIZE = detail::simd_size<T1>();
+    constexpr static std::size_t N_OUTPUT_SIMD_SIZE = detail::simd_size<T2>();
+
 public:
-    template<fair::meta::t_or_simd<T1> V>
-    [[nodiscard]] constexpr T2
-    process_one(V a) const noexcept {
-        return a;
+    fair::graph::work_return_t
+    work() noexcept {
+        using namespace stdx;
+        auto &out_port = output_port<"out">(this);
```

Comment:
Not a big thing, but I'd prefer first dealing with availability of readers, then with writers - instead commands from both being interleaved

Reply:
Has been rewritten by @mattkretz and his patch commits that I incorporated in this PR.